From 7660fee94b0ff8d630d67089a95287e48b9568db Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Sat, 19 Dec 2020 01:23:12 +0100
Subject: [PATCH] etnaviv: Fix rework ZSA into a derived state

In case the stencil is modified, it is also enabled. That was the
behavior of the original code, which was also the correct behavior,
so reinstate the behavior.

Fixes dEQP-GLES2.functional.fragment_ops.depth_stencil.* on STM32MP1 GC400T.

Upstream-Status: Posted [https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/8174]
Fixes: b29fe26d438 ("etnaviv: rework ZSA into a derived state")
Signed-off-by: Marek Vasut <marex@denx.de>
---
 src/gallium/drivers/etnaviv/etnaviv_zsa.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/gallium/drivers/etnaviv/etnaviv_zsa.c b/src/gallium/drivers/etnaviv/etnaviv_zsa.c
index 66d0c290c9f..b0f94993622 100644
--- a/src/gallium/drivers/etnaviv/etnaviv_zsa.c
+++ b/src/gallium/drivers/etnaviv/etnaviv_zsa.c
@@ -77,11 +77,13 @@ etna_zsa_state_create(struct pipe_context *pctx,
       if (so->stencil[0].fail_op != PIPE_STENCIL_OP_KEEP ||
           so->stencil[0].zfail_op != PIPE_STENCIL_OP_KEEP ||
           so->stencil[0].zpass_op != PIPE_STENCIL_OP_KEEP) {
+         cs->stencil_enabled = 1;
          cs->stencil_modified = 1;
       } else if (so->stencil[1].enabled) {
          if (so->stencil[1].fail_op != PIPE_STENCIL_OP_KEEP ||
              so->stencil[1].zfail_op != PIPE_STENCIL_OP_KEEP ||
              so->stencil[1].zpass_op != PIPE_STENCIL_OP_KEEP) {
+            cs->stencil_enabled = 1;
             cs->stencil_modified = 1;
          }
       }
-- 
2.29.2

